File: src/php/ArithmeticsBenchmark.php
<?php

namespace EntityFX\NetBenchmark\Core\Generic {

    use EntityFX\NetBenchmark\Core\Writer;
    
    class ArithmeticsBenchmark  extends ArithmeticsBase {

        protected function BenchImplementation()
        {
            parent::$R = 0;
            for ($i = 0; $i < $this->Iterrations; $i++)
            {
                self::$R += self::DoArithmetics($i);
            }
            return self::$R;
        }
    }
}

File: src/lua/generic/arithmeticsBenchmark.lua
ArithemticsBenchmark = class(ArithmeticsBase, function(a, writer, printToConsole)
    ArithmeticsBase.init(a, writer, printToConsole)
    a.name = "ArithemticsBenchmark"
end)

function ArithemticsBenchmark:benchImplementation()
    local R = 0
    for i=1,self.iterrations do
        R = R + self:doArithmetics(i)
    end
    return R
end